crates/ruff_server/src/server/api/notifications/did_change.rs

```diff
@@ -40,6 +41,12 @@ impl super::SyncNotificationHandler for DidChange {
             .make_mut()
             .apply_changes(content_changes, new_version, encoding);
 
+        // Publish diagnostics if the client doesnt support pull diagnostics
+        if !session.resolved_client_capabilities().pull_diagnostics {
+            let snapshot = session.take_snapshot(&uri).unwrap();
```

Comment:
We can `.unwrap()` here instead of throwing an error because the document is guaranteed to exist (we've already modified it) which means that the snapshot should also exist.

Reply:
I'd add it as a `SAFETY:` comment for posterity.